internal/pkg/template/diff/lcs.go

```diff
@@ -0,0 +1,53 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+type eqFunc func(inA, inB int) bool
+
+func longestCommonSubsequence[T any](a []T, b []T, eq eqFunc) ([]int, []int) {
```

Comment:
maybe an example input and output would be more helpful? from the function name i would assume it output the longest common sequence.